cdap-security/src/main/java/co/cask/cdap/security/securestore/FileSecureStoreProvider.java

```diff
@@ -0,0 +1,543 @@
+       Unfortunately that is not the observed behavior. */
+    if (ioe.getCause() instanceof UnrecoverableKeyException) {
+      return true;
+    }
+    // Workaround
+    return (ioe.getCause() == null)
+      && (ioe.getMessage() != null)
+      && ((ioe.getMessage().contains("Keystore was tampered")) || (ioe
+      .getMessage().contains("password was incorrect")));
+  }
+
+  private static void renameOrFail(Path src, Path dest) throws IOException {
+    Files.move(src, src.resolveSibling(dest));
+  }
+
+  private static String constructMetadataKey(String name) {
+    return name + METADATA_SUFFIX;
+  }
+
+  /**
+   * Initialize the keyStore.
+   *
+   * @throws IOException If there is a problem reading or creating the keystore.
+   */
+  private void locateKeystore() throws IOException {
+    Path oldPath = constructOldPath(path);
+    Path newPath = constructNewPath(path);
+    try {
+      keyStore = KeyStore.getInstance(SCHEME_NAME);
+      if (Files.exists(path)) {
+        // If the main file exists then the new path should not exist.
+        // Both existing means there is an inconsistency.
+        if (Files.exists(newPath)) {
+          throw new IOException(
+            String.format("Secure Store not loaded due to an inconsistency "
+                            + "('%s' and '%s' should not exist together)!!", path, newPath));
+        }
+        tryLoadFromPath(path, oldPath);
+      } else if (!tryLoadIncompleteFlush(newPath, oldPath)) {
+        // We were not able to load an existing key store. Create a new one.
+        keyStore.load(null, password);
+        LOG.info("New Secure Store initialized successfully.");
+      }
+    } catch (KeyStoreException | CertificateException | NoSuchAlgorithmException e) {
+      LOG.error("Can't create Secure Store. " + e);
+    }
+  }
+
+  /**
+   * Try loading from the user specified path, if that fails for any reason
+   * other than bad password then try loading from the backup path.
+   *
+   * @param path       Path to load from
+   * @param backupPath Backup path (_OLD)
+   * @throws IOException
+   */
+  private void tryLoadFromPath(Path path, Path backupPath) throws IOException {
+    try {
+      loadFromPath(keyStore, path, password);
+      // Successfully loaded the keystore. No need to keep the old file.
+      Files.deleteIfExists(backupPath);
+      LOG.info("Secure store loaded successfully.");
+    } catch (IOException ioe) {
+      // Try the backup path if the loading failed for any reason other than incorrect password.
+      if (!isBadOrWrongPassword(ioe)) {
+        // Mark the current file as CORRUPTED
+        renameOrFail(path, Paths.get(path.toString() + "_CORRUPTED_" + System.currentTimeMillis()));
+        // Try loading from the backup path
+        loadFromPath(keyStore, backupPath, password);
+        renameOrFail(backupPath, path);
+        LOG.warn(String.format("Secure store loaded successfully from '%s' since '%s'" + " was corrupted.",
```

Comment:
You don't need to use String.format with LOG.warn.
```java
LOG.warn("Secure store loaded successfully from '{}' since '{}' was corrupted.",
                  backupPath, path, e);
```